alacritty_terminal/src/input.rs

```diff
@@ -587,40 +605,62 @@ impl<'a, A: ActionContext + 'a> Processor<'a, A> {
 
         let button_changed = self.ctx.mouse().last_button != button;
 
+        let report_modes =
+            TermMode::MOUSE_REPORT_CLICK | TermMode::MOUSE_DRAG | TermMode::MOUSE_MOTION;
+        let report_mode_intersection = self.ctx.terminal().mode().intersects(report_modes);
+        let side = self.ctx.mouse().cell_side;
+
         self.ctx.mouse_mut().click_state = match self.ctx.mouse().click_state {
             ClickState::Click
                 if !button_changed && elapsed < self.mouse_config.double_click.threshold =>
             {
                 self.ctx.mouse_mut().block_url_launcher = true;
-                self.on_mouse_double_click(button, point);
+                if self.is_expand_selection(button, modifiers, report_mode_intersection) {
```

Comment:
We start new selections in `on_mouse_double_click`, why should we expand it here instead of inside `on_mouse_double_click`? This is the same for `on_mouse_triple_click`.